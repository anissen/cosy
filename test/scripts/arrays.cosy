{
	var x = [5, 6, 7]
	print x

	for e in x {
		print e
	}
}

{
	for e in [7,8,9] {
		print e
	}
}

{
	fn f(a Array Num) {
		print a
	}
	f([4,8,12])
	var y = [13,16]
	f(y)
}

{
	mut array = [6, 7]
	print 'array: '
	print array
	print 'array.push(8): '
	array.push(8)
	print array
	var p = array.push
	print 'array.length: ' + array.length
	p(9)
	print 'array.push(9): '
	print array
	print 'array.length: ' + array.length
	print 'concat([10, 11]): '
	array.concat([10, 11])
	print array
	print 'array.pop(): '
	array.pop()
	print array
	print 'array[2]: '
	print array[2]
	print 'array[3]: '
	print array[3]
}

{
	fn do_array_of_fn(functions) {
		for ff in functions {
			ff()
		}
	}
	do_array_of_fn([fn() { print '1' }, fn() { print '2' }])
}

{
	var arr = ['5','6','7']
	fn array_push(a Array Str) {
        print a
	}
	array_push(arr)
	print arr
}

{
    fn print_letter(l Str) {
        print l
    }
    for a in ['h', 'e', 'y'] {
        print_letter(a)
    }
}

{
	mut arr = [3, 4, 5]
	print 'arr: ' + arr
	print 'arr[0]: ' + arr[0]
	print 'arr[1]: ' + arr[1]
	print 'arr[2]: ' + arr[2]
	arr[0] = 6
	print 'arr[0] = 6: ' + arr[0]
	arr[1] += 45
	print 'arr[1] += 45: ' + arr[1]
	arr[1] -= 25
	print 'arr[1] -= 25: ' + arr[1]
	print 'arr: ' + arr
}
