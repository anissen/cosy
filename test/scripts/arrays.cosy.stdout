[5,6,7]
5
6
7
7
8
9
[4,8,12]
[13,16]
array: 
[6,7]
array.push(8): 
[6,7,8]
array.length: 3
array.push(9): 
[6,7,8,9]
array.length: 4
concat([10, 11]): 
[6,7,8,9,10,11]
array.pop(): 
[6,7,8,9,10]
array[2]: 
8
array[3]: 
9
1
2
[5,6,7]
[5,6,7]
h
e
y
arr: 3,4,5
arr[0]: 3
arr[1]: 4
arr[2]: 5
arr[0] = 6: 6
arr[1] += 45: 49
arr[1] -= 25: 24
arr: 6,24,5
